File: projects/engineering.purdue.edu/scriptingwithobjects/swocode/chap15/BroadcastServer.py
#!/usr/bin/env python

### BroadcastServer.py

import sys, socket, time                                             #(A)

try:                                                                 #(B)
    server_sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)   #(C)   
    server_sock.bind( ('', 0) )                                      #(D)
except socket.error, (value, message):                               #(E)
    if server_sock:                                                  #(F)
        server_sock.close()                                          #(G)
    print "No UDP server socket: %s" % message                       #(H)
    sys.exit(1)                                                      #(I)

server_sock.setsockopt(socket.SOL_SOCKET, socket.SO_BROADCAST, True) #(J)
dest = '<broadcast>', 9000                                           #(K)

msg_id = 1                                                           #(L)
while 1:                                                             #(M)
    msg_broadcast = "[%s]  Hark! The king hath arrived!\n" % msg_id  #(N)
    print "broadcasting message of id: %s" % msg_id                  #(O)
    server_sock.sendto( msg_broadcast, dest )                        #(P)
    time.sleep( 5 )                                                  #(Q)
    msg_id += 1                                                      #(R)

File: projects/engineering.purdue.edu/scriptingwithobjects/swocode/chap15/BroadcastServer.pl
#!/usr/bin/perl -w
use strict;

### BroadcastServer.pl

use IO::Socket;                                                      #(A)

my $sock = IO::Socket::INET->new( Proto => 'udp' );                  #(B)
die "No UDP Server Socket: $@\n" unless $sock;                       #(C)
$sock->sockopt( SO_BROADCAST() => 1 ) or die "sockopt: $!";          #(D)

my $dest = sockaddr_in( 9000, inet_aton( '255.255.255.255' ) );      #(E)
my $msg_broadcast;                                                   #(F)
my $msg_id = 1;                                                      #(G)
while (1) {                                                          #(H)
    $msg_broadcast = "[$msg_id]  Hark! The king hath arrived!\n";    #(I)
    print "broadcasting message of id: $msg_id\n";                   #(J)
    send( $sock, $msg_broadcast, 0, $dest ) or die "send(): $!";     #(K)
    sleep( 5 );                                                      #(L)
    $msg_id++;                                                       #(M)
}
$sock->close;                                                        #(N)
